persistent/Database/Persist/Types/Base.hs

```diff
@@ -1,6 +1,9 @@
 {-# LANGUAGE CPP #-}
 {-# LANGUAGE DeriveDataTypeable #-}
 {-# LANGUAGE OverloadedStrings #-}
+#if __GLASGOW_HASKELL__ >= 861
+{-# LANGUAGE DerivingVia #-}
+#endif
```

Comment:
As cool as this feature is, I'd rather not have CPP if we're going to need the boilerplate-y instance anyway.